src/rgw/rgw_rest_s3.cc

```diff
@@ -104,11 +104,13 @@ void rgw_get_errno_s3(rgw_http_error *e , int err_no)
   rgw_http_errors::const_iterator r = rgw_http_s3_errors.find(err_no);
 
   if (r != rgw_http_s3_errors.end()) {
-    e->http_ret = r->second.first;
-    e->s3_code = r->second.second;
+    e->http_ret = std::get<0>(r->second);
+    e->s3_code = std::get<1>(r->second);
+    e->message = std::get<2>(r->second);
```

Comment:
if an op has already set `s->err.message`, this would overwrite it

in most cases (all?), ops that set `s->err.message` do so before their call to `set_req_state_err()`

Reply:
not sure if i follow, i can't see that this touches req_state (s) anywhere and is just a helper function to get an error by number a populate the struct referenced by a pointer (e)...?

but i agree that this might need more testing unless we have a lot of functional testing that verifies error messages returned

Reply:
sorry, you're right about `rgw_get_errno_s3()`

`search_err()` from rgw_common.cc is the one that would overwrite `s->err.message` when called by `set_req_state_err(req_state* s, int err_no)`